app/test/unit/git/squash-test.ts

```diff
@@ -101,47 +101,50 @@ describe('git/cherry-pick', () => {
 
   it('squashes multiple commit non-sequential commits (reorders, non-conflicting)', async () => {
     const firstCommit = await makeSquashCommit(repository, 'first')
-    const secondCommit = await makeSquashCommit(repository, 'second')
-    await makeSquashCommit(repository, 'third')
-    const fourthCommit = await makeSquashCommit(repository, 'fourth')
+    await makeSquashCommit(repository, 'second')
+    const thirdCommit = await makeSquashCommit(repository, 'third')
+    await makeSquashCommit(repository, 'fourth')
+    const fifthCommit = await makeSquashCommit(repository, 'fifth')
 
     // From oldest to newest, log looks like:
     // - initial commit
     // - 'first' commit
     // - 'second' commit
     // - 'third' commit
     // - 'fourth' commit
+    // - 'fifth' commit
 
-    // Squashing 'second' and 'fourth' onto 'first'
-    // Thus, reordering 'fourth' to be before 'third'
+    // Squashing 'first' and 'fifth' onto 'third'
+    // Thus, reordering to 'second', 'first - third - fifth', 'fourth'
     const result = await squash(
       repository,
-      [secondCommit, fourthCommit],
-      firstCommit,
+      [firstCommit, fifthCommit],
+      thirdCommit,
       initialCommit.sha,
-      'Test Summary\n\nTest Body'
+      ''
     )
 
     expect(result).toBe(RebaseResult.CompletedWithoutError)
 
     // From oldest to newest, log should look like:
     // - initial commit - log[2]
-    // - the squashed commit 'Test Summary' - log[1]
-    // - 'third' commit - log[0]
+    // - the squashed commit 'first third fifth` - order by log history
```

Comment:
Before this refactor, the squashed commit would have been 'third first fifth' as third was the specified squashOnto commit. Now the test correctly verifies that it maintains the log history in the operation.

Reply:
Edit: Updated the test to send in toSquash to be in opposite log order to show that squash is smart enough to order it correctly.